app/main.py

```diff
@@ -1,3 +1,70 @@
+import datetime
+import json
+
+from app.calc_dist import dist_from_customer, calculate_cost_of_ride
+
+from app.calc_purchases import \
+    calc_purchases,\
+    shop_products,\
+    print_bought_products
+
+
 def shop_trip():
-    # write your code here
-    pass
+    with open("app/config.json", "r") as file:
+        config = json.load(file)
+        fuel_price = config["FUEL_PRICE"]
+        customers = config["customers"]
+        shops = config["shops"]
+
+    for customer in customers:
+        print(f"{customer['name']} has {customer['money']} dollars")
+        result = {}
+        for shop in shops:
+            distance = dist_from_customer(
+                customer["location"][0],
+                customer["location"][1],
+                shop["location"][0],
+                shop["location"][1])
+
+            ride_cost = calculate_cost_of_ride(
+                customer["car"]["fuel_consumption"],
+                distance,
+                fuel_price)
+
+            sum_of_products = calc_purchases(
+                customer["product_cart"],
+                shop["products"])
+
+            cost_of_trip = ride_cost * 2 + sum_of_products
+            cost_of_trip = round(cost_of_trip, 2)
+            result[shop["name"]] = cost_of_trip
+            print(
+                f"{customer['name']}'s "
+                f"trip to the {shop['name']} costs {cost_of_trip}")
+
+        ls_with_values_of_each_trip = sorted(list(result.values()))
+
+        ls_with_name_of_cheapest_trp_shop = \
+            [k for k, v in result.items()
```

Comment:
what are k and v?

Reply:
key and value in result dict

Reply:
it's uninformative names for variables, rewrite this